src/OpenIddict.Abstractions/Primitives/OpenIddictExtensions.cs

```diff
@@ -1798,6 +1792,15 @@ public static ClaimsPrincipal SetInternalAuthorizationId([NotNull] this ClaimsPr
         public static ClaimsPrincipal SetInternalTokenId([NotNull] this ClaimsPrincipal principal, string identifier)
             => principal.SetClaim(Claims.Private.TokenId, identifier);
 
+        /// <summary>
+        /// Sets the token usage associated with the claims principal.
+        /// </summary>
+        /// <param name="principal">The claims principal.</param>
+        /// <param name="tokenUsage">The token usage to store.</param>
+        /// <returns>The claims principal.</returns>
+        public static ClaimsPrincipal SetTokenUsage([NotNull] this ClaimsPrincipal principal, string tokenUsage)
```

Comment:
I'm tempted to say adding this extension is not super useful, but I'm all ears if you have any particular scenario in mind.

Reply:
I've just added this for consistency since we have Get*** method for this but don't have Set*** method. And again there is a test for Set*** method in old project. If it's not needed in new version - let's remove it.

Reply:
Makes sense. Since it's likely not very useful, let's remove it.

Reply:
Done